src/appsettings/settings.py

```diff
@@ -1210,3 +1212,89 @@ def check(self):
+            min_length (int): minimum length of the iterable (included).
+            max_length (int): maximum length of the iterable (included).
+            empty (bool): whether empty iterable is allowed. Deprecated in favor of min_length.
+        """
+        super(NestedSetting, self).__init__(*args, **kwargs)
+        warnings.warn("Checkers are deprecated in favor of validators.", DeprecationWarning)
+
+
+class NestedListSetting(IterableSetting):
+    """Nested list setting."""
+
+    def __init__(self, inner_setting, *args, **kwargs):
+        """
+        Initialization method.
+
+        Args:
+            inner_setting (Setting): setting that should be applied to list items.
+                NestedDictSetting is not supported at the moment.
+            name (str): the name of the setting.
+            default (dict): default value given to the setting.
+            required (bool): whether the setting is required or not.
+            prefix (str):
+                the setting's prefix (overrides ``AppSettings.Meta`` prefix).
+            call_default (bool): whether to call the default (if callable).
+            transform_default (bool): whether to transform the default value.
+            validators (list of callables): list of additional validators to use.
+            key_type: the type of the dict keys.
+            value_type (type): the type of dict values.
+            min_length (int): minimum length of the iterable (included).
+            max_length (int): maximum length of the iterable (included).
+            empty (bool): whether empty iterable is allowed. Deprecated in favor of min_length.
+        """
+        super(NestedListSetting, self).__init__(*args, **kwargs)
+        if not inner_setting.name:
+            inner_setting.name = self.name
+        inner_setting.parent_setting = self
+        self.inner_setting = inner_setting
+
+    def get_value(self):
+        try:
+            value = self.raw_value
+        except (AttributeError, KeyError) as err:
+            self._reraise_if_required(err)
+            default_value = self.default_value
+            if self.transform_default:
+                return self.transform(default_value)
+            return default_value
+        else:
+            return_value = []
+            for index, item in enumerate(value):
+                self.inner_setting.nested_list_index = index
+                return_value.append(self.inner_setting.get_value())
+            return tuple(return_value)
+
+    def check(self):
+        """Check the nested list setting itself and all its items."""
+        super(NestedListSetting, self).check()
+        try:
+            value = self.raw_value
+        except (AttributeError, KeyError) as err:
+            self._reraise_if_required(err)
+        else:
+            for index, item in enumerate(value):
+                self.inner_setting.nested_list_index = index
+                self.inner_setting.check()
```

Comment:
Could you raise all errors at once by concatenating them in a list/str?

Reply:
Oh yes, sure, I forgot that it applies to this case as well.

Reply:
Fixed (I'm collecting the errors in the same way as in `NestedDictSetting`).